### DISCUSSION TITLE
Change Simulation Mode

### DISCUSSION THREAD

**Discussion by dishant1602** [FPGA-Research/FABulous#255] [Q&A] [answered]
I have generated an eFPGA fabric and want to simulate it in Serial Mode which according to documentation is Mode 0. But I can't understand like what parameter/variable to change so that I am operating in Serial Mode(Mode 0).

**Comment by IAmMarcelJung** [comment, marked as answer]
Hi @dishant1602,

there currently is just code for the parallel configuration in the testbench. If you have own code for simulating in serial mode, then you should be able to use it without giving any parameter/variable. The fabric can automatically switch between the modes, depending on where it receives data. If (for whatever reason) data is received through multpile ports the priority is:
 1. UART
 2. Bitbang
 3. Parallel

This is all defined in `$name_of_your_project/Fabric/eFPGA_Config.v` if you want to take a closer look.

Hope this helps!
